stores/bench_test.go

```diff
@@ -200,51 +187,132 @@ func BenchmarkPruneSlabs(b *testing.B) {
+func BenchmarkObjects(b *testing.B) {
+	db, err := newTestDB(context.Background(), b.TempDir())
 	if err != nil {
-		return nil, err
-	} else if bucketID, err = res.LastInsertId(); err != nil {
-		return nil, err
+		b.Fatal(err)
 	}
 
-	stmt, err := db.Prepare(context.Background(), "INSERT INTO objects (created_at,object_id, db_bucket_id, size, mime_type, etag) VALUES (?, ?, ?, ?, '', '')")
+	// test parameters
+	objects := int(1e2)
+	bucket := "bucket"
+
+	// prepare database
+	dirs, err := insertObjects(db.DB(), bucket, objects)
 	if err != nil {
-		return nil, err
+		b.Fatal(err)
 	}
-	defer stmt.Close()
 
-	var path string
-	seen := make(map[string]struct{})
-	for i := 0; i < n; i++ {
-		for {
-			path = generateRandomPath(6)
-			if _, used := seen[path]; !used {
-				break
+	b.Run("Objects", func(b *testing.B) {
+		b.ResetTimer()
+		for i := 0; i < b.N; i++ {
+			if err := db.Transaction(context.Background(), func(tx sql.DatabaseTx) error {
+				_, err := tx.Objects(context.Background(), bucket, dirs[i%len(dirs)], "", "/", "", "", "", -1, object.EncryptionKey{})
+				return err
+			}); err != nil {
+				b.Fatal(err)
 			}
 		}
-		seen[path] = struct{}{}
+	})
 
-		size := frand.Intn(1e3)
-		if frand.Intn(10) == 0 {
-			path += "/"
-			size = 0
-			dirs = append(dirs, path)
+	// start rename benchmark
+	b.Run("RenameObjects", func(b *testing.B) {
+		b.ResetTimer()
+		for i := 0; i < b.N; i++ {
+			if err := db.Transaction(context.Background(), func(tx sql.DatabaseTx) error {
+				err := tx.RenameObjects(context.Background(), bucket, dirs[frand.Intn(i+1)%len(dirs)], dirs[frand.Intn(i+1)%len(dirs)], true)
+				if err != nil && !errors.Is(err, api.ErrObjectNotFound) {
+					return err
+				}
+				return nil
+			}); err != nil {
+				b.Fatal(err)
+			}
 		}
-		_, err := stmt.Exec(context.Background(), time.Now(), path, bucketID, size)
-		if err != nil {
-			return nil, err
+	})
+}
+
+// BenchmarkPrunableContractRoots benchmarks diffing the roots of a contract
+// with a given set of roots to determine which roots are prunable.
+//
+// 15.32 MB/s | M1 Max | cd32fad7 (diff ~2TiB of contract data per second)
+func BenchmarkPrunableContractRoots(b *testing.B) {
+	// define parameters
+	batchSize := int64(25600) // 100GiB of contract data
+	contractSize := 1 << 40   // 1TiB contract
+	sectorSize := 4 << 20     // 4MiB sector
+	numSectors := contractSize / sectorSize
+
+	// create database
+	db, err := newTestDB(context.Background(), b.TempDir())
+	if err != nil {
+		b.Fatal(err)
+	}
+
+	// prepare database
+	hk := types.PublicKey{1}
+	err = insertHost(db.DB(), hk)
+	if err != nil {
+		b.Fatal(err)
+	}
+
+	fcid := types.FileContractID{1}
+	roots, err := insertContract(db.DB(), hk, fcid, numSectors)
+	if err != nil {
+		b.Fatal(err)
+	}
+
+	// prepare batch
+	frand.Shuffle(len(roots), func(i, j int) {
+		roots[i], roots[j] = roots[j], roots[i]
+	})
+	batch := roots[:batchSize]
+
+	// start benchmark
+	b.ResetTimer()
+	b.SetBytes(batchSize * 32)
```

Comment:
I'd update this to sector size. MB of roots is not a very useful metric but being able to e.g. identify 1TB of prunable contract data makes more sense.